docs/cart-and-checkout-blocks/custom-inner-checkout-block.md

```diff
@@ -0,0 +1,203 @@
+
+The documentation for the `parent` property can be found on [developer.wordpress.org](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#parent).
+
+### Implement the frontend and editor components
+
+Now that we've defined the block metadata and registered the block, we need to implement the frontend and editor components. The frontend component is handled in [src/js/shipping-workshop-block/block.js](https://github.com/woocommerce/wceu23-shipping-workshop-final/blob/main/src/js/shipping-workshop-block/block.js) while the editor component is handled in [src/js/shipping-workshop-block/edit.js](https://github.com/woocommerce/wceu23-shipping-workshop-final/blob/main/src/js/shipping-workshop-block/edit.js).
+
+## Checkout block injection & UI Integration
+
+💡 The checkout block injection and UI integration process involves registering the custom inner block within the Checkout block, managing its local state and connecting it to the checkout lifecycle.
+
+### Register the block with `registerCheckoutBlock`
+
+Now that we've created our custom inner block, we need to register it within the Checkout block. This is done using `registerCheckoutBlock()` from the `@woocommerce/blocks-checkout` package. We are doing this in [src/js/shipping-workshop-block/frontend.js](https://github.com/woocommerce/wceu23-shipping-workshop-final/blob/main/src/js/shipping-workshop-block/frontend.js):
+
+```js
+registerCheckoutBlock( {
+	metadata,
+	component: Block,
+} );
+```
+
+The documentation for `registerCheckoutBlock()` can be found on [github.com](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/blocks/packages/checkout/blocks-registry/README.md#registercheckoutblock-options-).
+
+### Control the block visibility and placement
+
+The visibility and placement is controlled by the parent block. For example, if a custom inner block has the parent property `woocommerce/checkout-shipping-address-block`, but the shopper selected local pickup, then the shipping address block will not be visible. As a result of that, the custom inner block will not be visible either.
+
+### Manage the local block state using `useState`
+
+To manage the local block state, we can use the `useState` hook. This allows us to store and update the user input within the block. In our case, we are storing the selected option, e.g. `Try again another day`, and the user input if the shopper selected the option `Other`. This takes place in [the following section](https://github.com/woocommerce/wceu23-shipping-workshop-final/blob/1d65e685fb40c207e8e06d6c066b890f527276c5/src/js/shipping-workshop-block/block.js#L58-L62):
+
+```js
+const [
+	selectedAlternateShippingInstruction,
+	setSelectedAlternateShippingInstruction,
+] = useState( 'try-again' );
+const [ otherShippingValue, setOtherShippingValue ] = useState( '' );
+```
+
+The documentation for `useState` can be found on [react.dev](https://react.dev/docs/hooks-state).
```

Comment:
This section on `useState` feels a little misplaced in this context. Developers working with WooCommerce blocks should already be familiar with React fundamentals. Instead of explaining `useState`, perhaps this section could be repurposed to discuss local state vs. using the data stores. Again I think this really depends on the aim and purpose of the doc, if it's a specific tutorial about the "Not at home" block then it's fine to keep.

Reply:
Based on your [review](https://github.com/woocommerce/woocommerce/pull/56787#pullrequestreview-2725526590), I’ll create a more generic doc.